lib/reporters/tap.js

```diff
@@ -41,79 +39,42 @@ class TapReporter {
 	}
 
 	start() {
-		return 'TAP version 13';
+		return supertap.start();
 	}
 
 	test(test) {
-		const output = [];
-
-		let directive = '';
-		const passed = test.todo ? 'not ok' : 'ok';
-
-		if (test.todo) {
-			directive = '# TODO';
-		} else if (test.skip) {
-			directive = '# SKIP';
-		}
-
-		const title = stripAnsi(test.title);
-
-		const appendLogs = () => {
-			if (test.logs) {
-				test.logs.forEach(log => {
-					const logLines = indentString(log, 4);
-					const logLinesWithLeadingFigure = logLines.replace(
-						/^ {4}/,
-						'  * '
-					);
-
-					output.push(logLinesWithLeadingFigure);
-				});
-			}
-		};
-
-		output.push(`# ${title}`);
-
-		if (test.error) {
-			output.push(format('not ok %d - %s', ++this.i, title));
-			appendLogs();
-			output.push(dumpError(test.error, true));
-		} else {
-			output.push(format('%s %d - %s %s', passed, ++this.i, title, directive).trim());
-			appendLogs();
-		}
-
-		return output.join('\n');
+		return supertap.test(test.title, {
+			passed: !test.error,
+			index: ++this.i,
+			todo: test.todo,
+			skip: test.skip,
+			comment: test.logs,
+			error: test.error ? dumpError(test.error, true) : null
```

Comment:
`dumpError` doesn't return an `Error` instance, which `supertap`s API says is expected. Perhaps the API documentation should be loosened, or clearer on what properties are expected on non-`Error` objects?

Reply:
Agree, will clarify that.

Reply:
Done.